tests/test_models/test_tbats.py

```diff
@@ -126,3 +162,157 @@ def test_prediction_interval(model, example_tsds):
 @pytest.mark.parametrize("model", [TBATSModel(), BATSModel()])
 def test_save_load(model, example_tsds):
     assert_model_equals_loaded_original(model=model, ts=example_tsds, transforms=[], horizon=3)
+
+
+def test_forecast_decompose_not_fitted():
+    model = _TBATSAdapter(model=BATS())
+    with pytest.raises(ValueError, match="Model is not fitted!"):
+        model.forecast_components(horizon=5)
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_predict_components_not_implemented(periodic_ts, estimator):
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    with pytest.raises(NotImplementedError, match="Prediction decomposition isn't currently implemented!"):
+        for segment in test.columns.get_level_values("segment"):
+            model._models[segment].predict_components(horizon=3)
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_decompose_forecast_output_format(periodic_ts, estimator):
+    horizon = 5
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    for segment in test.columns.get_level_values("segment"):
+        components = model._models[segment]._decompose_forecast(horizon=horizon)
+        assert isinstance(components, np.ndarray)
+        assert components.shape[0] == horizon
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_named_components_output_format(periodic_ts, estimator):
+    horizon = 5
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    for segment in test.columns.get_level_values("segment"):
+        components = model._models[segment]._decompose_forecast(horizon=horizon)
+        named_components = model._models[segment]._named_components(raw_components=components)
+
+        for component in named_components.values():
+            assert len(component) == horizon
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+@pytest.mark.parametrize(
+    "data,params,message",
+    (
+        (
+            "sinusoid_ts",
+            {"use_box_cox": False, "use_trend": False, "use_arma_errors": False, "seasonal_periods": [7, 14]},
+            "Seasonal components is not fitted!",
+        ),
+        (
+            "periodic_ts",
+            {
+                "use_box_cox": True,
+                "use_trend": True,
+                "use_damped_trend": True,
+                "use_arma_errors": True,
+                "seasonal_periods": [7, 14],
+            },
+            "ARMA components is not fitted!",
+        ),
+    ),
+)
+def test_fitted_components_warnings(data, estimator, params, message, request):
```

Comment:
We can actualy use any fixture for this purpose, just replace the attributes of the fitted model as we need

Reply:
Right now it's obvious what behaviour we are testing, we are showing that components set may depend on data.

Reply:
If we change the internal estimation algorithm, this test should be irrelevant. Changes in other parts shouldn't impact this behaviour.

Reply:
We model the behaviour of estimator when user set some components but they were not fitted

Reply:
Reworked